test/unit/test_diskfile.py

```diff
@@ -207,40 +208,46 @@ def test_on_sproxyd_down(self, mock_ping=None):
         self.assertNotIn(('localhost', 81), sfs.sproxyd_hosts_set)
         self.assertEqual([], list(sfs.sproxyd_hosts))
 
-    @mock.patch('swift.common.bufferedhttp.http_connect_raw',
-                side_effect=lambda *args: eventlet.sleep(0.02))
+    @mock.patch('socket.socket.connect', side_effect=socket.timeout)
     def test_do_http_connection_timeout(self, mock_http_connect):
-        sfs = SproxydFileSystem({'sproxyd_conn_timeout': 0.01}, mock.Mock())
-        self.assertRaises(swift.common.exceptions.ConnectionTimeout,
-                          sfs._do_http, 'me', {}, 'HTTP_METH', '/')
-
-    def test_do_http_timeout(self):
-        class HTTPConn(object):
-
-            def getresponse(self):
-                eventlet.sleep(0.02)
+        timeout = 0.01
+        sfs = SproxydFileSystem({'sproxyd_conn_timeout': timeout}, mock.Mock())
 
-            def close(self):
-                pass
+        regex = r'^.*connect timeout=%s.*$' % timeout
+        self.assertRaisesRegexp(urllib3.exceptions.ConnectTimeoutError, regex,
+                                sfs._do_http, 'me', {}, 'HTTP_METH', '/')
 
-        sfs = SproxydFileSystem({'sproxyd_proxy_timeout': 0.01}, mock.Mock())
-
-        with mock.patch('swift.common.bufferedhttp.http_connect_raw',
-                        return_value=HTTPConn()):
-            self.assertRaises(eventlet.Timeout, sfs._do_http, 'me',
-                              {}, 'HTTP_METH', '/')
-
-    @mock.patch('swift.common.bufferedhttp.http_connect_raw',
-                return_value=FakeHTTPConn())
+    def test_do_http_timeout(self):
+        server1 = eventlet.listen(('127.0.0.1', 0))
+        (ip, port) = server1.getsockname()
+
+        def run_server1(sock):
+            (client, addr) = sock.accept()
+            eventlet.sleep(0.1)
+
+        t = eventlet.spawn(run_server1, server1)
+        timeout = 0.01
+        sfs = SproxydFileSystem({'sproxyd_host': '%s:%d' % (ip, port),
+                                 'sproxyd_proxy_timeout': timeout},
+                                mock.Mock())
+
+        regex = r'^.*read timeout=%s.*$' % timeout
+        self.assertRaisesRegexp(urllib3.exceptions.ReadTimeoutError, regex,
```

Comment:
Thinking of this: are all exceptions `urllib3` and co raise properly converted into Swift exceptions?